src/fmu/dataio/_metadata.py

```diff
@@ -54,7 +54,10 @@ def generate_meta_tracklog() -> list:
 def generate_meta_masterdata(config: dict) -> Optional[dict]:
     """Populate metadata from masterdata section in config."""
     if not config or "masterdata" not in config.keys():
```

Comment:
Suggest separating these two into separate warnings, as they are symptoms of different problems and require different actions to fix. Suggest first check for `config`, then check for `masterdata` with informative warning message.

(Perhaps config without `masterdata` should trigger error not warning...)